cloudbuild.yaml

```diff
@@ -0,0 +1,117 @@
+##
+# Required images
+#
+# In order for this to run you need to deploy the Pixite Android cloud builders.
+#
+# ```bash
+# git clone git@github.com:pixiteapps/android-cloud-build /tmp/android-cloud-build
+# cd /tmp/android-cloud-build
+# gcloud builds submit .
+# ```
+
+steps:
+
+# Update the build number
+- name: 'gcr.io/$PROJECT_ID/buildnum:1.0'
+  id: setup_env
+  waitFor: ['-']
+  args: ['gs://${_CONFIG_BUCKET}/buildnum']
+
+- name: gcr.io/cloud-builders/gcloud
+  id: generate_cache_key
+  entrypoint: bash
+  waitFor: ['-']
+  args:
+  - -c
+  - |
+    ./checksum.sh checksum.txt
+
+- name: gcr.io/$PROJECT_ID/restore_cache:1.0
+  id: restore_cache
+  waitFor: ['generate_cache_key']
+  args:
+  - --bucket=gs://${_CACHE_BUCKET}
+  - --key=gradle-$( checksum checksum.txt )
+
+  # Build the project
+- name: gcr.io/$PROJECT_ID/android:28
+  id: download_dependencies
+  args: ['./gradlew', 'dependencies']
+  <<: &env
+    env:
+    - 'TERM=dumb'
+    - 'GRADLE_USER_HOME=/workspace/.gradle'
+    # For standard maching, use these options
+    - 'GRADLE_OPTS="-Dorg.gradle.daemon=false -Dorg.gradle.workers.max=2 -Dkotlin.incremental=false"'
+    - 'JAVA_TOOL_OPTIONS="-Xmx1024m"'
+    # For High CPU machine, use these options
+    # - 'GRADLE_OPTS="-Dorg.gradle.daemon=false -Dkotlin.incremental=false"'
+    # - 'JAVA_TOOL_OPTIONS="-Xmx4g"'
+    - 'BRANCH_NAME=$BRANCH_NAME'
+    - 'ARTIFACT_BUCKET=gs://${_ARTIFACT_BUCKET}'
+  waitFor:
+  - setup_env
+  - restore_cache
+
+- name: 'gcr.io/$PROJECT_ID/android:28'
+  id: build
+  args: ['./gradlew', 'bundleDebug', 'assembleDebug', '-Ptivi.versioncode=1']
```

Comment:
This should be BUILD_NUM (I'll manually hack it remain consistent with the CircleCI build numbers)

Reply:
Agreed.  The reason I hard coded this was because the current builder doesn't `eval` the parameters so doesn't support env variables in the args array.  I'm planning to update the builder to support that, since I haven't had the need for environment variables in args until I integrated your repo.